plugins/inputs/x509_cert/java_key_store_test.go

```diff
@@ -0,0 +1,193 @@
+package x509_cert
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/pavlo-v-chernykh/keystore-go/v4"
+	"github.com/stretchr/testify/require"
+	"software.sslmate.com/src/go-pkcs12"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+type SelfSignedCert struct {
+	CertPEM []byte
+	KeyPEM  []byte
+	CertDER []byte
+}
+
+// generateTestKeystores creates temporary JKS & PKCS#12 keystores for testing
+func generateTestKeystores(t *testing.T) (pkcs12Path, jksPath string) {
+	t.Helper()
+
+	// Generate a test certificate
+	selfSigned := generateSelfSignedCert(t)
+
+	pkcs12Path = createTestPKCS12(t, selfSigned.CertPEM, selfSigned.KeyPEM)
+	jksPath = createTestJKS(t, selfSigned.CertDER)
+
+	return pkcs12Path, jksPath
+}
+
+// generateSelfSignedCert generates a dummy self-signed certificate
+func generateSelfSignedCert(t *testing.T) SelfSignedCert {
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName:   "Test Certificate",
+			Organization: []string{"Test Org"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(24 * time.Hour),
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		BasicConstraintsValid: true,
+	}
+
+	certDER, err := x509.CreateCertificate(rand.Reader, &template, &template, &privKey.PublicKey, privKey)
+	require.NoError(t, err)
+
+	return SelfSignedCert{
+		CertPEM: pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certDER}),
+		KeyPEM:  pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(privKey)}),
+		CertDER: certDER,
+	}
+}
+
+// createTestPKCS12 creates a temporary PKCS#12 keystore
+func createTestPKCS12(t *testing.T, certPEM, keyPEM []byte) string {
+	t.Helper()
+
+	// Decode certificate
+	block, _ := pem.Decode(certPEM)
+	if block == nil {
+		t.Fatal("failed to parse certificate PEM")
+	}
```

Comment:
Please use the power of `require` as this makes the test-purpose easier to read!
```suggestion
	require.NotNil(t, block, "failed to parse certificate PEM")
```
Same for other spots below.